### PR TITLE
Signature Aggregator ACP-118 updates

### PR DESCRIPTION
**Pull request by iansuvak**
~~This should pass the tests but not necessarily be merged into the PR until all of the required commits make it into default branches. Right now the main outstanding one would be the subnet-evm that references newer avalanchego commit that includes latest coreth changes.~~

The commits have been updated to reference default branches ones now. 

Includes a bugfix on the validation logic.

I'm open to suggestions on cleaner prefix packing and to avoid copying into `blsSignatureBuf` if anyone has ideas.

### CODE REVIEW

**Review by feuGeneA — COMMENTED** [review]
LGTM! Happy to see it working!

**Comment by geoff-vball** [comment]
I think you may have accidentally included an update to teleporter? The go.work.sum was added back

**Comment by iansuvak** [comment]
> I think you may have accidentally included an update to the `teleporter` submodule? The go.work.sum was added back

Thanks, I had teleporter updated to a branch commit that was not necessary so I downgraded it v1.0.4 which is what it is on `main` branch I believe https://github.com/ava-labs/awm-relayer/tree/main/tests/contracts/lib . I'm happy to bump to any version that we need but it wouldn't be related to this and the parent PR which shouldn't require changes to teleporter dependencies

**Review by geoff-vball — APPROVED** [review]

File: signature-aggregator/aggregator/aggregator.go [hunk 1]
```diff
@@ -459,8 +462,9 @@ func (s *SignatureAggregator) isValidSignatureResponse(
 		)
 		return blsSignatureBuf{}, false
 	}
-
-	return signature, true
+	blsSig := blsSignatureBuf{}
+	copy(blsSig[:], signature[:])
```
Comment (michaelkaplan13) [inline signature-aggregator/aggregator/aggregator.go:466]: Should we have a check higher up that the length of `signature` is the expected length?

File: signature-aggregator/api/api.go [hunk 2]
```diff
@@ -123,12 +123,13 @@ func signatureAggregationAPIHandler(logger logging.Logger, aggregator *aggregato
 			return
 		}
 
-		if isEmptyOrZeroes(message.Bytes()) || isEmptyOrZeroes(justification) {
+		if isEmptyOrZeroes(message.Bytes()) && isEmptyOrZeroes(justification) {
```
[the remark this answers is not in the document]
Comment (iansuvak) [reply signature-aggregator/api/api.go:126]: Filed #446

File: signature-aggregator/aggregator/aggregator.go (hunk 1 above)
Comment (iansuvak) [reply signature-aggregator/aggregator/aggregator.go:466]: Possibly, we do call `bls.SignatureFromBytes` and `bls.Verify` on it's result so presumably those wouldn't work if it wasn't correct lenght but an explicit check wouldn't hurt

Comment (iansuvak) [reply signature-aggregator/aggregator/aggregator.go:475]: Addressed in 811c97a

**Review by bernard-avalabs — COMMENTED** [review]
LGTM

**Pull request merged**